chalice/deploy/newdeployer.py

```diff
@@ -0,0 +1,472 @@
+function needs the ``role_arn`` that's the result of a previous ``create_role``
+API call, a ``Variable`` object is used to forward this information.
+
+The executor also records these variables with their associated resources so a
+``deployed.json`` file can be written to disk afterwards.  An ``APICall``
+takes an optional resource object when it's created whose ``resource_name``
+is used as the key in the ``deployed.json`` dictionary.
+
+
+"""
+import os
+
+from typing import List, Set, Dict, Any, Optional, Union  # noqa
+from botocore.session import Session  # noqa
+
+from chalice.utils import OSUtils, UI
+from chalice.deploy import models
+from chalice.config import Config  # noqa
+from chalice import app  # noqa
+from chalice.deploy.packager import LambdaDeploymentPackager
+from chalice.deploy.packager import PipRunner, SubprocessPip
+from chalice.deploy.packager import DependencyBuilder as PipDependencyBuilder
+from chalice.policy import AppPolicyGenerator
+from chalice.constants import LAMBDA_TRUST_POLICY
+from chalice.constants import DEFAULT_LAMBDA_TIMEOUT
+from chalice.constants import DEFAULT_LAMBDA_MEMORY_SIZE
+from chalice.awsclient import TypedAWSClient, ResourceDoesNotExistError
+
+
+def create_default_deployer(session):
+    # type: (Session) -> Deployer
+    client = TypedAWSClient(session)
+    osutils = OSUtils()
+    pip_runner = PipRunner(pip=SubprocessPip(osutils=osutils),
+                           osutils=osutils)
+    dependency_builder = PipDependencyBuilder(
+        osutils=osutils,
+        pip_runner=pip_runner
+    )
+    return Deployer(
+        resource_builder=ApplicationGraphBuilder(),
+        deps_builder=DependencyBuilder(),
+        build_stage=BuildStage(
+            steps=[
+                InjectDefaults(),
+                DeploymentPackager(
+                    packager=LambdaDeploymentPackager(
+                        osutils=osutils,
+                        dependency_builder=dependency_builder,
+                        ui=UI(),
+                    ),
+                ),
+                PolicyGenerator(
+                    policy_gen=AppPolicyGenerator(
+                        osutils=osutils
+                    ),
+                ),
+            ],
+        ),
+        plan_stage=PlanStage(
+            client=client, osutils=osutils,
+        ),
+        executor=Executor(client),
+    )
+
+
+class UnresolvedValueError(Exception):
+    MSG = (
+        "The API parameter '%s' has an unresolved value "
+        "of %s in the method call: %s"
+    )
+
+    def __init__(self, key, value, method_name):
+        # type: (str, models.Placeholder, str) -> None
+        msg = self.MSG % (key, value, method_name)
+        super(UnresolvedValueError, self).__init__(msg)
+        self.key = key
+        self.value = value
+        self.method_name = method_name
+
+
+class Deployer(object):
+    def __init__(self,
+                 resource_builder,  # type: ApplicationGraphBuilder
```

Comment:
Change this name to match its new type more closely? resource_builder seems pretty vague to me.

Reply:
Updated.